src/text_shaper/directwrite_shaper.cpp

```diff
@@ -616,9 +467,9 @@ void directwrite_shaper::shape(font_key _font,
 
 std::optional<rasterized_glyph> directwrite_shaper::rasterize(glyph_key _glyph, render_mode _mode)
 {
-    FontInfo const& fontInfo = d->fonts.at(_glyph.font);
-    IDWriteFontFace5* fontFace = fontInfo.fontFace.Get();
-    const float fontEmSize = _glyph.size.pt * (96.0 / 72.0);
+    DxFontInfo const& fontInfo = d->fonts.at(_glyph.font);
+    IDWriteFontFace5* fontFace = fontInfo.fontFace;
+    const float fontEmSize = d->ptToEm(_glyph.size.pt);
```

Comment:
I just realize that ptToEm is a member of `d` (Private). I think that doesn't have to be as it's just transforming the input without any dependency. Maybe make it a free function and put it into the anonymous namespace?

Reply:
yea it should be outside `Private`.